s2wav/Dimension helper functions.py

```diff
@@ -0,0 +1,373 @@
+        kernel (str): The wavelet type.
+        lam (float): Wavelet parameter which determines the scale factor between consecutive wavelet scales.
+        j (int): Wavelet scale.
+    
+    Returns:
+        Band-limit (int).
+    '''
+    if kernel.lower() == "s2dw":
+        return math.ceil(lam ** (j+1))
+    else:
+        #This code currently does not support other wavelet types.
+        return print("Invalid wavelet type.")
+
+
+
+def L0(kernel: str, lam: float, j: int) -> int:
+    '''Computes the minimum harmonic index supported by the given wavelet scale.
+
+    Args:
+        ernel (str): The wavelet type.
+        lam (float): Wavelet parameter which determines the scale factor between consecutive wavelet scales.
+        j (int): Wavelet scale.
+    
+    Returns:
+        el_min (int).
+    '''
+    if kernel.lower() == "s2dw":
+        return math.ceil(lam ** (j-1))
+    else:
+        #This code currently does not support other wavelet types
+        return print("Invalid wavelet type.")
+
+
+
+def j_max(L: int, lam: float) -> int:
+    '''Computes needlet maximum level required to ensure exact reconstruction.
+
+    Args:
+        L (int): Upper harmonic band-limit.
+        lam (float): Wavelet parameter which determines the scale factor between consecutive wavelet scales.
+    
+    Returns:
+        j_max (int).
+    '''
+    return math.ceil(log(L) / log(lam))
+
+
+
+
+
+#files originally from s2let_helper
+
+def n_phi(sampling_scheme: str, L: int) -> int:
+    '''Returns the number of phi samples.
+
+    Args:
+        sampling_scheme (str): Either S2LET_SAMPLING_MW or S2LET_SAMPLING_MW_SS.
+        L (int): Upper harmonic band-limit.
+
+    Returns:
+        nphi (int): Number of phi samples.
+    '''
+    if sampling_scheme.lower() == "mwss":
+        return sampling_mw_ss_nphi(L)
+    elif sampling_scheme.lower() == "mw":
+        return ampling_mw_nphi(L)
+
+
+def n_theta(sampling_scheme: str, L: int) -> int:
+    '''Returns the number of theta samples.
+
+    Args:
+        sampling_scheme (str): Either S2LET_SAMPLING_MW or S2LET_SAMPLING_MW_SS.
+        L (int): Upper harmonic band-limit.
+
+    Returns: 
+        ntheta (int): Number of theta samples.
+    '''
+    if sampling_scheme.lower() == "mwss":
+        return sampling_mw_ss_ntheta(L)
+    elif sampling_scheme.lower() == "mw":
+        return sampling_mw_ntheta(L)
+
+
+def n_px(sampling_scheme: str, L: int) -> int:
+    '''Returns the number of phi samples multiplied by the number of theta samples.
+    
+    Args:
+        sampling_scheme (str): Either S2LET_SAMPLING_MW or S2LET_SAMPLING_MW_SS.
+        L (int): Upper harmonic band-limit.
+    
+    Returns:
+        (int): nphi * ntheta.
+    '''
+    return n_phi(sampling_scheme, L) * n_theta(sampling_scheme, L)
+
+
+def n_lm(L: int) -> int:
+    '''Returns the square of the harmonic band-limit.
+    
+    Args:
+        L (int): Upper harmonic band-limit.
+
+    Returns:
+        Square of L (int).
```

Comment:
Returns should always be `data_type: description of data`, its quite nit picky but it will break auto-docs (which is a faff) if we aren't really strict with this

Reply:
Also the description for this should be something like "Total number of harmonic coefficients"